openedx/core/djangoapps/course_date_signals/tests.py

```diff
@@ -114,5 +115,225 @@ def test_sequence_with_graded_and_ungraded_assignments(self):
+        and default PLS do not assign due dates to non graded assignments.
+
+        If custom PLS is not set, the subsection will fall back to the default
+        PLS logic of evenly spaced sections.
+        """
+        course = self.courses[0]
+        chapter = course.get_children()[0]
+        with self.store.bulk_operations(course.id):
+            sequential = ItemFactory.create(category='sequential', parent=chapter, due_num_weeks=3)
+            ItemFactory.create(category='sequential', parent=chapter)
+            ItemFactory.create(category='sequential', parent=chapter)
+            expected_dates = [
+                (course.location, {}),
+                (chapter.location, timedelta(days=28)),
+                (sequential.location, {'due': timedelta(days=21)})
+            ]
+            course = modulestore().get_item(course.location)
+            self.assertCountEqual(extract_dates_from_course(course), expected_dates)
+
+    @override_waffle_flag(CUSTOM_PLS, active=True)
+    def test_extract_dates_from_course_custom_and_default_pls_one_subsection_graded(self):
+        """
+        A section with a subsection that has due_num_weeks and
+        a subsection without due_num_weeks that has graded content.
+        Default PLS should apply for the subsection without due_num_weeks that has graded content.
+
+        If custom PLS is not set, the subsection will fall back to the default
+        PLS logic of evenly spaced sections.
+        """
+        course = self.courses[0]
+        chapter1 = course.get_children()[0]
+        sequential1 = ItemFactory.create(category='sequential', parent=chapter1, due_num_weeks=2)
+        vertical1 = ItemFactory.create(category='vertical', parent=sequential1)
+        problem1 = ItemFactory.create(category='problem', parent=vertical1)
+
+        chapter2 = ItemFactory.create(category='chapter', parent=course)
+        sequential2 = ItemFactory.create(category='sequential', parent=chapter2, graded=True)
+        vertical2 = ItemFactory.create(category='vertical', parent=sequential2)
+        problem2 = ItemFactory.create(category='problem', parent=vertical2)
+
+        expected_dates = [
+            (course.location, {}),
+            (chapter1.location, timedelta(days=21)),
+            (sequential1.location, {'due': timedelta(days=14)}),
+            (vertical1.location, {'due': timedelta(days=14)}),
+            (problem1.location, {'due': timedelta(days=14)}),
+            (chapter2.location, timedelta(days=42)),
+            (sequential2.location, {'due': timedelta(days=42)}),
+            (vertical2.location, {'due': timedelta(days=42)}),
+            (problem2.location, {'due': timedelta(days=42)})
+        ]
+        course = modulestore().get_item(course.location)
+        with patch.object(utils, 'get_expected_duration', return_value=timedelta(weeks=6)):
+            actual = extract_dates_from_course(course)
```

Comment:
Whoops. Sorry! I meant to not assign the actual variable at all and just have the assert here